### PR TITLE
Mandate form bugs

### PR DESCRIPTION
**Pull request by yspringfield**
Fixed: #8763

### CODE REVIEW

File: libs/ui/src/lib/form/table/form-table.component.ts [hunk 1]
```diff
@@ -83,7 +87,8 @@ export class FormTableComponent<T> implements OnInit, OnDestroy {
   }
 
   add() {
-    this.formItem = this.form.createControl({});
+    const deafultValue = this.defaultFormValue ? this.defaultFormValue : null;
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/ui/src/lib/form/table/form-table.component.ts:90]: This should work
```suggestion
    const deafultValue = this.defaultFormValue || null;
```

File: firestore.rules [hunk 2]
```diff
@@ -390,7 +390,10 @@ service cloud.firestore {
       allow create: if (userOrgId() == getContract(incomingData().contractId).buyerId
         || userOrgId() == getContract(incomingData().contractId).sellerId)
         && notInMaintenance();
-      allow update: if false;
+      allow update:if ('mandate' == getContract(incomingData().contractId).type
+        || userOrgId() == getContract(incomingData().contractId).sellerId)
```
[the remark this answers is not in the document]
Comment (yspringfield) [reply firestore.rules:394]: Yes, that's the whole point of the functionality. To permit sellers to update/create their terms without having to import.

File: libs/ui/src/lib/form/table/form-table.component.ts [hunk 3]
```diff
@@ -44,13 +45,16 @@ export class FormViewDirective { }
 export class FormTableComponent<T> implements OnInit, OnDestroy {
   @Input() columns: Record<string, string> = {};
   @Input() form: FormList<T>;
+  @Input() defaultFormValue: T;
```
[the remark this answers is not in the document]
Comment (yspringfield) [reply libs/ui/src/lib/form/table/form-table.component.ts:48]: It's used [here.](https://github.com/blockframes/blockframes/pull/8775/files/efa3f8ae523034fc868ded02120b13fdcd2b3040#diff-3beea8856516094d7a18f2177b4df350d9ef22366aef7a16e5394495c66713f4R91)

File: libs/ui/src/lib/form/table/form-table.component.ts [hunk 4]
```diff
@@ -83,7 +87,7 @@ export class FormTableComponent<T> implements OnInit, OnDestroy {
   }
 
   add() {
-    this.formItem = this.form.createControl({});
+    this.formItem = this.form.createControl(this.defaultFormValue || null);
```
Comment (phpgeekfr) [inline libs/ui/src/lib/form/table/form-table.component.ts:90]: before your PR it was initialized with ` this.formItem = this.form.createControl({})` 

using `null` is the same ?

Comment (yspringfield) [reply libs/ui/src/lib/form/table/form-table.component.ts:90]: It's not the same, thanks for noticing it.

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**